python/lib/dcm2bids_imaging_pipeline_lib/base_pipeline.py

```diff
@@ -78,109 +75,127 @@ def __init__(self, loris_getopt_obj, script_name):
-                    tarchive_id = self.imaging_upload_obj.imaging_upload_dict["TarchiveID"]
-                    self.dicom_archive = try_get_dicom_archive_with_id(self.env.db, tarchive_id)
-                    if self.dicom_archive is not None:
-                        success = True
-                    else:
-                        err_msg += f"Could not load tarchive dictionary for TarchiveID {tarchive_id}"
+            mri_upload = try_get_mri_upload_with_id(self.env.db, upload_id)
+            if mri_upload is None:
+                log_error_exit(
+                    self.env,
+                    f"Did not find an entry in mri_upload associated with 'UploadID' {upload_id}",
+                    lib.exitcode.SELECT_FAILURE,
+                )
+
+            self.mri_upload = mri_upload
+            if self.mri_upload.dicom_archive is None:
+                log_error_exit(
+                    self.env,
+                    f"Did not find a DICOM archive associated with upload ID {upload_id}",
+                    lib.exitcode.SELECT_FAILURE,
+                )
+
+            self.dicom_archive = self.mri_upload.dicom_archive
 
         elif tarchive_path:
             archive_location = tarchive_path.replace(self.dicom_lib_dir, "")
-            self.dicom_archive = try_get_dicom_archive_with_archive_location(self.env.db, archive_location)
-            if self.dicom_archive is not None:
-                success, new_err_msg = self.imaging_upload_obj.create_imaging_upload_dict_from_tarchive_id(
-                    self.dicom_archive.id
+            dicom_archive = try_get_dicom_archive_with_archive_location(self.env.db, archive_location)
+            if dicom_archive is None:
+                log_error_exit(
+                    self.env,
+                    f"Could not load tarchive dictionary for ArchiveLocation {archive_location}",
+                    lib.exitcode.SELECT_FAILURE,
                 )
 
-                if not success:
-                    err_msg += new_err_msg
-            else:
-                err_msg += f"Could not load tarchive dictionary for ArchiveLocation {archive_location}"
-
-        if not success and not self.force:
-            log_error_exit(self.env, err_msg, lib.exitcode.SELECT_FAILURE)
+            self.dicom_archive = dicom_archive
+
+            mri_uploads = self.dicom_archive.mri_uploads
+            match mri_uploads:
+                case []:
+                    log_error_exit(
+                        self.env,
+                        f"Did not find an entry in mri_upload associated with 'TarchiveID' {self.dicom_archive.id}",
+                        lib.exitcode.SELECT_FAILURE,
+                    )
+                case [mri_upload]:
+                    self.mri_upload = mri_upload
+                case _:
+                    log_error_exit(
+                        self.env,
+                       f"Found {len(mri_uploads)} rows in mri_upload for 'TarchiveID' {self.dicom_archive.id}",
```

Comment:
might be worth printing the different UploadIDs matching to not have to query it later when debugging.